### PR TITLE
chore(protocol): update protocol configurations for alpha-2 testnet

### PR DESCRIPTION
**Pull request by dantaik**
- [X] chainId: 167,
- [X] maxNumBlocks: 61, // owner:daniel
- [X] blockHashHistory: 40, // owner:daniel
- [X] zkProofsPerBlock: 1, // owner:daniel
- [X] maxVerificationsPerTx: 10, //owner:david. Each time one more block is verified, there will be ~20k more gas cost
- [X] commitConfirmations: 0, // owner:daniel
- [X] maxProofsPerForkChoice: 3, // owner:daniel
- [X] blockMaxGasLimit: 6000000, // owner:david
- [X] maxTransactionsPerBlock: 79, //  owner:david
- [X] maxBytesPerTxList: 120000, // owner:david 
- [x] minTxGasLimit: 21000, // owner:david
- [x] anchorTxGasLimit: 250000, // owner:david
- [X] slotSmoothingFactor: 16789, // owner:daniel
- [X] rewardBurnBips: 100, // owner:daniel. 100 basis points or 1%
- [X] proposerDepositPctg: 25, // owner:daniel - 25%
- [X] feeBaseMAF: 1024,
- [X] blockTimeMAF: 1024,
- [X] proofTimeMAF: 1024,
- [X] rewardMultiplierPctg: 400, //  owner:daniel - 400%
- [X] feeGracePeriodPctg: 200, // owner:daniel - 200%
- [X] feeMaxPeriodPctg: 400, // owner:daniel - 400%
- [X] blockTimeCap: 60 seconds, // owner:daniel
- [X] proofTimeCap: 30 minutes, // owner:daniel
- [X] bootstrapDiscountHalvingPeriod: 30 days, // owner:daniel
- [X] initialUncleDelay: 10 minutes,
- [X] enableTokenomics: true,
- [X] enablePublicInputsCheck: true,
- [X] enableProofValidation: true

### CODE REVIEW

**Review by davidtaikocha — DISMISSED** [review]

**Comment by davidtaikocha** [comment]
There are some other configuration files in `k8s-config` when deploying our different networks. So this `LibSharedConfig.sol` file is not used.

File: packages/protocol/contracts/libs/LibSharedConfig.sol [hunk 1]
```diff
@@ -16,20 +16,25 @@ library LibSharedConfig {
...
-                // the 'the maximum value of the multiplier' close to 20.0
+                // Assuming proof time 1 hour, block time 10 seconds
+                // Level of parallelization is 60*60/10 = 360
+                // We give it extra 50% more slots, then we have 360*1.5+1=
+                maxNumBlocks: 541,
+                // Changed it to a smaller value
+                // Assuming block time 10 seconds, we support cross-chain tx
+                // to have 1 confirmation within 600 seconds (10 minutes).
+                // Then we have: 600/10 = 60.
+                blockHashHistory: 60,
```
Comment (Brechtpd) [inline packages/protocol/contracts/libs/LibSharedConfig.sol:27]: Are these values set to these low values for testing purposes so they get triggered faster in the testnet?

Comment (dantaik) [reply packages/protocol/contracts/libs/LibSharedConfig.sol:27]: Yes. Especially this one `blockHashHistory`.  `maxNumBlocks' should be much bigger in mainnet I believe.

File: packages/protocol/contracts/libs/LibSharedConfig.sol [hunk 2]
```diff
@@ -16,20 +16,25 @@ library LibSharedConfig {
         return
             TaikoData.Config({
                 chainId: 167,
-                maxNumBlocks: 2049, // up to 2048 pending blocks
-                blockHashHistory: 100000,
-                // This number is calculated from maxNumBlocks to make
-                // the 'the maximum value of the multiplier' close to 20.0
+                // Assuming proof time 1 hour, block time 10 seconds
+                // Level of parallelization is 60*60/10 = 360
```
Comment (davidtaikocha) [inline packages/protocol/contracts/libs/LibSharedConfig.sol:20]: If the proof time is 1 hour, then a block will need `proofTime` (1 hour) + `UncleProofDelay (state.avgProofTime ≈ 1 hour)` ≈ 2 hours before getting verified?

File: packages/protocol/contracts/libs/LibSharedConfig.sol [hunk 3]
```diff
@@ -8,42 +8,57 @@ pragma solidity ^0.8.9;
...
-                anchorTxGasLimit: 250000,
-                feePremiumLamda: 590,
-                rewardBurnBips: 100, // 100 basis points or 1%
-                proposerDepositPctg: 25, // 25%
+                maxNumBlocks: 61, // owner:daniel
+                blockHashHistory: 40, // owner:daniel
+                zkProofsPerBlock: 1, // owner:daniel
+                maxVerificationsPerTx: 20, //owner:david - TODO: what's the actual tx gas cost?
+                commitConfirmations: 0, // owner:daniel
+                maxProofsPerForkChoice: 3, // owner:daniel
```
Comment (davidtaikocha) [inline packages/protocol/contracts/libs/LibSharedConfig.sol:39]: not sure will we also increase this value? seems it means right now we only allow 3 fastest honest provers?

**Comment by davidtaikocha** [comment]
Changed `maxVerificationsPerTx` to `10`. According to the test, each time one more block is verified, there will be ~20k more gas cost, and the base cost of the verification process is ~90K, so i think verifying extra 10 blocks (~25K extra gas cost) is more acceptable than 20 blocks(~45K extra gas cost) when proposing / proving blocks:
```
{
  method: 'verifyBlock',
  nums: 1,
  gasUsed: BigNumber { value: "92590" }
}
{
  method: 'verifyBlock',
  nums: 2,
  gasUsed: BigNumber { value: "85522" }
}
{
  method: 'verifyBlock',
  nums: 3,
  gasUsed: BigNumber { value: "105052" }
}
{
  method: 'verifyBlock',
  nums: 4,
  gasUsed: BigNumber { value: "127941" }
}
{
  method: 'verifyBlock',
  nums: 5,
  gasUsed: BigNumber { value: "149151" }
}
{
  method: 'verifyBlock',
  nums: 6,
  gasUsed: BigNumber { value: "168680" }
}
{
  method: 'verifyBlock',
  nums: 7,
  gasUsed: BigNumber { value: "191570" }
}
{
  method: 'verifyBlock',
  nums: 8,
  gasUsed: BigNumber { value: "212780" }
}
{
  method: 'verifyBlock',
  nums: 9,
  gasUsed: BigNumber { value: "232309" }
}
{
  method: 'verifyBlock',
  nums: 10,
  gasUsed: BigNumber { value: "255199" }
}
```

**Comment by davidtaikocha** [comment]
Updated `maxTransactionsPerBlock` to `80`, since its the upper limit of @smtmfft's current circuits, and left `blockMaxGasLimit` as it is (5M), which is a safe number for @smtmfft's current circuits.

**Comment by davidtaikocha** [comment]
For `maxBytesPerTxList`, currently there is a discussion in discord: https://discord.com/channels/984015101017346058/985743259802402856/1074904085326745780 , maybe we can not do much about this, but set a value which is a little bit smaller than 128K?

File: packages/protocol/contracts/libs/LibSharedConfig.sol (hunk 3 above)
Comment (dantaik) [reply packages/protocol/contracts/libs/LibSharedConfig.sol:39]: Only 3 for now.

**Comment by davidtaikocha** [comment]
> For `maxBytesPerTxList`, currently there is a discussion in discord: https://discord.com/channels/984015101017346058/985743259802402856/1074904085326745780 , maybe we can not do much about this, but set a value which is a little bit smaller than 128K?

Update `maxBytesPerTxList` to 120KB, since 128KB is the upper limit of a geth transaction, so using 120KB for the L1 transaction list calldata, 8K for the remaining tx fields.

This means we may have maximun 3~4 contract creations (24KB size limit), or 20 bridges (~6KB), or 79 transfers (maxTransactionsPerBlock) in one L2 block.

**Comment by dantaik** [comment]
@davidtaikocha could you please add some of the above rationals as comments in the file. I don't mind the file is large with a lot of comments.

**Comment by davidtaikocha** [comment]
> @davidtaikocha could you please add some of the above rationals as comments in the file. I don't mind the file is large with a lot of comments.

Sure, there may be still some small adjustments these days, and i will add more comments in the file after that.

**Comment by davidtaikocha** [comment]
> @davidtaikocha could you please add some of the above rationals as comments in the file. I don't mind the file is large with a lot of comments.

Sure, there may be still some small adjustments these days, and i will add more comments in the file after that.

**Comment by dantaik** [comment]
@davidtaikocha Mark this one as draft and let me know when comments are added so we can all take another look.

**Comment by davidtaikocha** [comment]
> @davidtaikocha Mark this one as draft and let me know when comments are added so we can all take another look.

Comments added.

File: packages/protocol/contracts/libs/LibSharedConfig.sol [hunk 4]
```diff
@@ -8,42 +8,57 @@ pragma solidity ^0.8.18;
...
-                proposerDepositPctg: 25, // 25%
+                maxNumBlocks: 61, // owner:daniel
+                blockHashHistory: 40, // owner:daniel
+                zkProofsPerBlock: 1, // owner:daniel
+                maxVerificationsPerTx: 10, //owner:david. Each time one more block is verified, there will be ~20k more gas cost.
+                commitConfirmations: 0, // owner:daniel
+                maxProofsPerForkChoice: 3, // owner:daniel
+                blockMaxGasLimit: 6000000, // owner:david. Set it to 6M, since its the upper limit of the Alpha-2 testnet's circuits.
+                maxTransactionsPerBlock: 79, //  owner:david. Set it to 79  (+1 TaikoL2.anchor transaction = 80), and 80 is the upper limit of the Alpha-2 testnet's circuits.
+                maxBytesPerTxList: 120000, // owner:david. Set it to 120KB, since 128KB is the upper size limit of a geth transaction, so using 120KB for the proposed transactions list calldata, 8K for the remaining tx fields.
```
Comment (Brechtpd) [inline packages/protocol/contracts/libs/LibSharedConfig.sol:42]: Are we planning on creating invalid blocks? If so are we sure we are able to prove this many bytes in our invalid tx list function on L2 within the 6M block gas limit? If we won't create invalid blocks then all good.

Comment (davidtaikocha) [reply packages/protocol/contracts/libs/LibSharedConfig.sol:42]: yes in alpha-2, there will be only one whitelisted proposer, so i think this 120K config is fine for now.

**Review by davidtaikocha — DISMISSED** [review]

**Review by davidtaikocha — APPROVED** [review]

**Review by Brechtpd — APPROVED** [review]

**Pull request merged**